batcher/aligned-sdk/src/sdk.rs

```diff
@@ -86,22 +85,33 @@ pub async fn submit_multiple_and_wait_verification(
     max_fees: &[U256],
     wallet: Wallet<SigningKey>,
     nonce: U256,
-) -> Result<Vec<AlignedVerificationData>, errors::SubmitError> {
-    let aligned_verification_data = submit_multiple(
+) -> Vec<Result<AlignedVerificationData, errors::SubmitError>> {
+    let mut aligned_verification_data = submit_multiple(
         batcher_url,
         network,
         verification_data,
         max_fees,
         wallet,
         nonce,
     )
-    .await?;
+    .await;
 
+    // TODO: open issue: use a join to .await all at the same time, avoiding the loop
+    // And await only once per batch, no need to await multiple proofs if they are in the same batch.
+    let mut error_awaiting_batch_verification: Option<errors::SubmitError> = None;
     for aligned_verification_data_item in aligned_verification_data.iter() {
-        await_batch_verification(aligned_verification_data_item, eth_rpc_url, network).await?;
+        if let Ok(aligned_verification_data_item) = aligned_verification_data_item {
+            if let Err(e) = await_batch_verification(aligned_verification_data_item, eth_rpc_url, network).await {
+                error_awaiting_batch_verification = Some(e);
+                break;
+            }
+        }
+    }
+    if let Some(error_awaiting_batch_verification) = error_awaiting_batch_verification {
+        aligned_verification_data.push(Err(error_awaiting_batch_verification));
     }
```

Comment:
IMO, we should return a `Vec<Result<AlignedVerificationData>>` containing all the proofs sent by the user. However, `Ok()` should indicate that the verification data was actually verified, not simply sent. In the case of an `Err()`, we should include additional information about the `VerificationData`. For example, if the error is a timeout, the user could decide whether to verify the batch inclusion manually.

Reply:
This could be addressed in another PR, since it falls out of scope of this one.
Please make an issue of it before resolving this conversation

Reply:
Created #1393!